TestLoadLargeRules is flaky

## Flaky Test
### Which jobs are failing
```text
{"level":"warn","ts":"2026-03-12T15:52:30.525421+0800","caller":"etcdserver/util.go:170","msg":"apply request took too long","took":"124.859275ms","expected-duration":"100ms","prefix":"","request":"header:<ID:4525164215846883341 > put:<key:\"/pd/0/region_label/test_4844\" value_size:234 >","response":"size:5"}
{"level":"info","ts":"2026-03-12T15:52:30.525721+0800","caller":"traceutil/trace.go:171","msg":"trace[490458128] put","detail":"{key:/pd/0/region_label/test_4844; req_size:267; response_revision:4845; }","duration":"147.304769ms","start":"2026-03-12T15:52:30.378332+0800","end":"2026-03-12T15:52:30.525637+0800","steps":["trace[490458128] 'process raft request'  (duration: 22.132178ms)","trace[490458128] 'marshal mvccpb.KeyValue'  (duration: 124.398704ms)"],"step_count":2}
{"level":"info","ts":"2026-03-12T15:52:30.731723+0800","caller":"traceutil/trace.go:171","msg":"trace[2075406135] put","detail":"{key:/pd/0/region_label/test_4845; req_size:267; response_revision:4846; }","duration":"154.89274ms","start":"2026-03-12T15:52:30.576706+0800","end":"2026-03-12T15:52:30.731598+0800","steps":["trace[2075406135] 'process raft request'  (duration: 132.16049ms)","trace[2075406135] 'marshal mvccpb.KeyValue'  (duration: 22.41421ms)"],"step_count":2}
panic: test timed out after 5m0s
	running tests:
		TestLoadLargeRules (5m0s)
goroutine 6025 [running]:
testing.(*M).startAlarm.func1()
	/usr/local/go/src/testing/testing.go:2682 +0x605
created by time.goFunc
	/usr/local/go/src/time/sleep.go:215 +0x45
goroutine 1 [chan receive, 5 minutes]:
```
### CI link
https://prow.tidb.net/view/gs/prow-tidb-logs/pr-logs/pull/tikv_pd/10339/pull-unit-test-next-gen-1/2031997832684113920
### Reason for failure (if possible)

### Anything else
test=TestLoadLargeRules signatures=TIMEOUT_PANIC,PANIC

--- a/pkg/mcs/scheduling/server/rule/watcher_test.go
+++ b/pkg/mcs/scheduling/server/rule/watcher_test.go
@@ -98,6 +98,7 @@ func prepare(t require.TestingT) (context.Context, *clientv3.Client, func()) {
 	re.NoError(err)
 	<-etcd.Server.ReadyNotify()
 
+	ops := make([]clientv3.Op, 0, etcdutil.MaxEtcdTxnOps)
 	for i := 1; i < rulesNum+1; i++ {
 		rule := &labeler.LabelRule{
 			ID:       "test_" + strconv.Itoa(i),
@@ -108,7 +109,15 @@ func prepare(t require.TestingT) (context.Context, *clientv3.Client, func()) {
 		value, err := json.Marshal(rule)
 		re.NoError(err)
 		key := keypath.RegionLabelKeyPath(rule.ID)
-		_, err = clientv3.NewKV(client).Put(ctx, key, string(value))
+		ops = append(ops, clientv3.OpPut(key, string(value)))
+		if len(ops) == etcdutil.MaxEtcdTxnOps {
+			_, err = client.Txn(ctx).Then(ops...).Commit()
+			re.NoError(err)
+			ops = ops[:0]
+		}
+	}
+	if len(ops) > 0 {
+		_, err = client.Txn(ctx).Then(ops...).Commit()
 		re.NoError(err)
 	}
 

